webapp/channels/src/components/admin_console/server_logs/logs.tsx

```diff
@@ -92,6 +94,23 @@ export default class Logs extends React.PureComponent<Props, State> {
         this.setState({page: this.state.page - 1});
     };
 
+    downloadLogs = async () => {
```

Comment:
Instead of going to all the work of downloading and stringifying the logs here, what if we were to actually add a `download=true` parameter to the `getLogs` API and then have the download button trigger that API with the download parameter set along with whatever the page normally calls to load the logs? To do that, you have to set some headers on the API response, but there's a `web.WriteFileResponse` helper function on the server to handle all that for you.

The APIs for file attachments and some other things behave similarly, and it feels like it would be a more proper way of doing this since we're relying mostly on standard HTTP headers and browser behaviour to process the logs instead of doing it in JS which seems like it could cause problems when the logs are larger on a busier server. The server could even potentially stream the results, but that feels like that might require more work which is outside the scope of this.

Thoughts on all this? CCing @hanzei as well since this would be a pretty big change in the approach

Reply:
Yeah good point. This is an appropriate way tbh. As when logs will be larger it will be heavy to process on frontend. I can make this change. If you are raising a new issue related to this, you can assign it to me directly. Or I can do in this issue only.

Reply:
2/5 that adding a separate `/logs/download` endpoint makes sense as the other `/logs/` endpoint do some kind of parsing on the logs.